### PR TITLE
Deduplicator cleanup; refactoring; and tests

### PR DESCRIPTION
**Pull request by darinkrauss**
@jh-bate It may be easier to review it by commit. Each commit is very focused and make incremental changes.

### CODE REVIEW

File: data/deduplicator/delegate/delegate.go [hunk 1]
```diff
@@ -13,25 +13,30 @@ package delegate
...
-	}
-}
-
 type factory struct {
 	factories []deduplicator.Factory
 }
 
+func NewFactory(factories []deduplicator.Factory) (deduplicator.Factory, error) {
+	if len(factories) == 0 {
+		return nil, app.Error("delegate", "factories is missing")
```
Comment (jh-bate) [inline data/deduplicator/delegate/delegate.go]: missing or empty? prob doesn't matter

Comment (darinkrauss) [inline data/deduplicator/delegate/delegate.go]: Yes, I actually had two separate checks `factories == nil` and `len(factories) == 0`, but decided they were essentially the same thing (ie. no factories provided) so I just combined the two.

**Comment by darinkrauss** [comment]
@jh-bate Any other comments on this one?

**Comment by jh-bate** [comment]
LGTM

File: data/store/mongo/config_test.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package mongo_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	// . "github.com/onsi/gomega"
+)
+
+var _ = PDescribe("Config", func() {
```
Comment (jh-bate) [inline data/store/mongo/config_test.go]: will we have tests?

File: dataservices/dataservices.go [hunk 3]
```diff
@@ -166,8 +181,8 @@ func initializeUserServicesClient(configLoader config.Loader, logger log.Logger)
 	return userServicesClient, nil
 }
 
-func initializeAPI(configLoader config.Loader, logger log.Logger, dataStore store.Store, userServicesClient client.Client, reporter version.Reporter) (server.API, error) {
-	return api.NewStandard(logger, dataStore, userServicesClient, reporter)
+func initializeAPI(configLoader config.Loader, logger log.Logger, dataStore store.Store, dataDeduplicatorFactory deduplicator.Factory, userServicesClient client.Client, reporter version.Reporter) (server.API, error) {
```
Comment (jh-bate) [inline dataservices/dataservices.go]: `configLoader` not used?

**Comment by darinkrauss** [comment]
Rebased on to master with final commit 91a9317ec892a4509a37e28cc679d405b19b36ad. Closed.

**Pull request closed without merging**